src/cmd/services/m3dbnode/config/fs.go

```diff
@@ -78,11 +88,110 @@ type FilesystemConfiguration struct {
 
 	// ForceIndexSummariesMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceIndexSummariesMmapMemory bool `yaml:"force_index_summaries_mmap_memory"`
+	ForceIndexSummariesMmapMemory *bool `yaml:"force_index_summaries_mmap_memory"`
 
 	// ForceBloomFilterMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceBloomFilterMmapMemory bool `yaml:"force_bloom_filter_mmap_memory"`
+	ForceBloomFilterMmapMemory *bool `yaml:"force_bloom_filter_mmap_memory"`
+}
+
+// FilePathPrefixOrDefault returns the configured file path prefix if configured, or a
+// default value otherwise.
+func (f FilesystemConfiguration) FilePathPrefixOrDefault() string {
+	if f.FilePathPrefix != nil {
+		return *f.FilePathPrefix
+	}
+
+	return defaultFilePathPrefix
+}
+
+// WriteBufferSizeOrDefault returns the configured write buffer size if configured, or a
+// default value otherwise.
+func (f FilesystemConfiguration) WriteBufferSizeOrDefault() int {
+	if f.WriteBufferSize != nil && *f.WriteBufferSize > 1 {
```

Comment:
Is it ok to silently ignore bad values if they've been set? Similar comment to a bunch of other places where it's ignored

Reply:
yeah I was kinda torn on this one. I can't do validation on the pointer types AND allow them to not be set which makes this really tricky. I could add a `Validate` method to the struct, but thats confusing also because we're using the `validator` package on these structs...I figured ignoring really dumb values wouldn't be the end of the world. Thoughts?

Reply:
Can we at least log that we're using a default value?

Reply:
I spoke to @arnikola and I'm just gonna add a manual validate method